src/renderer/components/cluster-settings/components/cluster-node-shell-setting.tsx

```diff
@@ -0,0 +1,140 @@
+import { FormSwitch, Switcher } from "../../switch";
+
+interface Props {
+  cluster: Cluster;
+}
+
+@observer
+export class ClusterNodeShellSetting extends React.Component<Props> {
+  @observable nodeShellImage = "";
+  @observable imagePullSecret = "";
+  @observable usePrivateRegistry = false;
+
+  constructor(props: Props) {
+    super(props);
+    makeObservable(this);
+  }
+
+  componentDidMount() {
+    this.usePrivateRegistry = !!this.props.cluster.getImagePullSecret();
+
+    disposeOnUnmount(this,
+      autorun(() => {
+        this.nodeShellImage = this.props.cluster.getNodeShellImage();
+        this.imagePullSecret = this.props.cluster.getImagePullSecret();
+      })
+    );
+  }
+
+  onImageChange = (value: string) => {
+    this.nodeShellImage = value;
+  };
+
+  onSecretChange = (value: string) => {
+    this.imagePullSecret = value;
+  };
+
+  saveImage = () => {
+    this.props.cluster.preferences.nodeShellImage = this.nodeShellImage;
+  };
+
+  saveSecret = () => {
+    this.props.cluster.preferences.imagePullSecret = this.imagePullSecret;
+  };
+
+  resetImage = () => {
+    this.nodeShellImage = initialNodeShellImage; //revert to default
+  };
+
+  clearSecret = () => {
+    this.imagePullSecret = "";
+  };
+
+  onSwitchChange = () => {
+    this.usePrivateRegistry = !this.usePrivateRegistry;
+
+    if (!this.usePrivateRegistry) {
+      this.imagePullSecret = "";
+      this.saveSecret();
+    }
+  };
+
+  render() {
+    const resetImageIcon = <Icon small material="close" onClick={this.resetImage} tooltip={"Reset"}/>;
+    const clearSecretIcon = <Icon small material="close" onClick={this.clearSecret} tooltip={"Clear"}/>;
+
+    return (
+      <>
+        <section>
+          <SubTitle title="Node shell image" id="node-shell-image"/>
+          <Input
+            theme="round-black"
+            value={this.nodeShellImage}
+            onChange={this.onImageChange}
+            onBlur={this.saveImage}
+            iconRight={resetImageIcon}
+          />
+          <small className="hint">
+            Node shell image. Used for creating node shell pod.
+          </small>
+        </section>
+        <section>
+          <FormSwitch
+            control={
+              <Switcher
+                checked={this.usePrivateRegistry}
+                onClick={this.onSwitchChange}
+                name="private-registry"
+              />
+            }
+            label="Use private registry"
+          />
+        </section>
+        {this.usePrivateRegistry &&
+        <section>
+          <SubTitle title="Image pull secret" id="image-pull-secret"/>
+          <Input
+            theme="round-black"
+            value={this.imagePullSecret}
+            onChange={this.onSecretChange}
+            onBlur={this.saveSecret}
+            iconRight={clearSecretIcon}
+          />
+          <small className="hint">
+            Image pull secret. Used for pulling image from a private registry.
```

Comment:
This should specify that it is the name of a pre-existing secret object.

Reply:
Is it better? 
<img width="776" alt="Screen Shot 2021-07-12 at 4 45 16 PM" src="https://user-images.githubusercontent.com/38247153/125290288-35c5f300-e331-11eb-9608-8dc698af98a9.png">

Reply:
Yes thanks.